code/modules/power/cell.dm

```diff
@@ -501,6 +501,25 @@
 			icon_state = "mfc-empty"
 	. = ..()
 
+// Crafted Microfusion cell - large energy weapons
+/obj/item/stock_parts/cell/ammo/mfc/bad
+	name = "shoddy microfusion cell"
+	desc = "A microfusion cell, typically used as ammunition for large energy weapons. This one looks a little dubious though."
+	icon = 'icons/fallout/objects/powercells.dmi' //TODO: give these bad icons
+	icon_state = "mfc-full"
+	maxcharge = 1000
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/stock_parts/cell/ammo/mfc/update_icon()
+	switch(charge)
+		if (501 to 1000)
```

Comment:
`		if (501 to 1000)`
to
`		if (501 to maxcharge)`

Reply:
Poked it and yeah it's not updating the children's icons like it should. I'd say just deal with the bad cells flashing they're half-full!

Reply:
Alright, changed everything to a math based system and now none of the cells are updating properly.